doc/concept/build_system.md

```diff
@@ -0,0 +1,73 @@
+_Note: This document is designed to render nicely with GitHub-flavored Markdown.  My apologies in advance to everyone else._
+
+Build System Concepts
+=====================
+
+Goals of this document
+----------------------
+
+Be deleberate about choosing a build system; explain criteria, options, choices, and decisions.
+
+Strategy to choose a build system is:
+1. Enumerate good experiences to retain
+2. Enumerate bad experiences to avoid
+3. Be judiciously terse; the longer this is, the fewer people will read it
+4. Enumerate & experiment with options
+5. Make decision
+
+
+Definitions
+-----------
+
+**Contributor:** Who is a contributor?  Is there some base-level knowledge we can expect?
```

Comment:
Contributor is someone who is sending a pull request.

Maintainer is a set of people that have had a track record to contribution, has volunteered to help with the project maintenance (tagging releases, updating the website, managing infrastructure, etc.).

Code Owner is a person who has responsibility for a sub-part of the overall project, and has approval rights to merge changes from non-code-owner contributors.